### PR TITLE
UCS/SYS: optimization for namespace processing

### PR DESCRIPTION
**Pull request by hoopoepg**
## What
UCS/SYS: optimization for namespace processing

## Why ?
in case of high PPN reading of system namespace may take
  significant time. as workaround of issue added caching
  of values

## How ?
added caching of namespace values

### CODE REVIEW

File: src/ucs/sys/sys.c [hunk 1]
```diff
@@ -48,17 +48,23 @@
 #define UCS_PROCESS_NS_FIRST       0xF0000000U
 #define UCS_PROCESS_NS_NET_DFLT    0xF0000080U
 
+#define UCS_NS_INFO_ITEM(_id, _name, _dflt) \
+    [_id] = {.name = _name, .dflt = _dflt, \
+             .init_once = UCS_INIT_ONCE_INITIALIZER}
```
Comment (yosefe) [inline src/ucs/sys/sys.c:53]: why need both dflt and value?
let's have only value, and init it the the default

Comment (hoopoepg) [reply src/ucs/sys/sys.c:53]: default value is used in another function:
```
int ucs_sys_ns_is_default(ucs_sys_namespace_type_t ns)
{
    return ucs_sys_get_ns(ns) == ucs_sys_namespace_info[ns].dflt;
}
```

Comment (hoopoepg) [reply src/ucs/sys/sys.c:53]: ```dflt``` value may be declared as ```const``` and ```value``` may be pre-initialized by default value

Comment (yosefe) [reply src/ucs/sys/sys.c:53]: ok

**Review by yosefe — APPROVED** [review]

File: src/ucs/sys/sys.c [hunk 2]
```diff
@@ -48,17 +48,23 @@
 #define UCS_PROCESS_NS_FIRST       0xF0000000U
 #define UCS_PROCESS_NS_NET_DFLT    0xF0000080U
 
+#define UCS_NS_INFO_ITEM(_id, _name, _dflt) \
+    [_id] = {.name = _name, .dflt = _dflt, .value = _dflt, \
```
Comment (yosefe) [inline src/ucs/sys/sys.c:52]: pls add ( ) around params usage

File: src/ucs/sys/sys.c [hunk 3]
```diff
@@ -1329,15 +1335,17 @@ ucs_sys_ns_t ucs_sys_get_ns(ucs_sys_namespace_type_t ns)
...
+        snprintf(filename, sizeof(filename), "%s/%s", UCS_PROCESS_NS_DIR,
+                 ucs_sys_namespace_info[ns].name);
 
-    res = stat(filename, &st);
-    if (res == 0) {
-        return (ucs_sys_ns_t)st.st_ino;
+        res = stat(filename, &st);
+        if (res == 0) {
+            ucs_sys_namespace_info[ns].value = (ucs_sys_ns_t)st.st_ino;
+        }
```
Comment (yosefe) [inline src/ucs/sys/sys.c:1345]: pls add ucs_debug(stat(%s): %m) if failed

**Comment by hoopoepg** [comment]
will squash after tests are completed

File: src/ucs/sys/sys.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucs/sys/sys.c]: added () around params

File: src/ucs/sys/sys.c (hunk 3 above)
Comment (hoopoepg) [reply src/ucs/sys/sys.c:1347]: added debug output

**Review by yosefe — APPROVED** [review]

**Pull request merged**